### PR TITLE
Make Shopping cart and ticket with dao and service

### PR DESCRIPTION
**Pull request by vicire**

### CODE REVIEW

File: src/main/java/cinema/Main.java [hunk 1]
```diff
@@ -42,12 +43,13 @@ public static void main(String[] args) {
 
         AuthenticationService service = (AuthenticationService) injector
                 .getInstance(AuthenticationService.class);
-        service.register("Human", "qwerty");
-        try {
-            System.out.println(service.login("Human", "qwerty"));
-            System.out.println(service.login("Human", "qwerty1"));
```
Comment (Max-Pochepets) [inline src/main/java/cinema/Main.java]: did you stop testing your login? you'r gonna merge this to master

File: src/main/java/cinema/dao/ShoppingCartDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public ShoppingCart getByUser(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            return session.createQuery("select sc from ShoppingCart sc "
+                    + "left join fetch sc.tickets t "
```
Comment (Max-Pochepets) [inline src/main/java/cinema/dao/ShoppingCartDaoImpl.java]:
```suggestion
                    + "left join fetch sc.tickets "
```
redundant alias

**Review by Max-Pochepets — APPROVED** [review]

File: src/main/java/cinema/Main.java (hunk 1 above)
Comment (vicire) [reply src/main/java/cinema/Main.java]: moved it back

File: src/main/java/cinema/dao/ShoppingCartDaoImpl.java (hunk 2 above)
Comment (vicire) [reply src/main/java/cinema/dao/ShoppingCartDaoImpl.java]: fixed

File: src/main/java/cinema/service/ShoppingCartServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    private ShoppingCartDao shoppingCartDao;
+    @Inject
+    private TicketDao ticketDao;
+
+    @Override
+    public void addSession(MovieSession movieSession, User user) {
+        Ticket ticket = new Ticket();
+        ticket.setMovieSession(movieSession);
+        ticket.setUser(user);
+        ticketDao.add(ticket);
```
Comment (KseniiaMakarova) [inline src/main/java/cinema/service/ShoppingCartServiceImpl.java]: if there is no shopping cart for the user - we still save ticket to the db, is it ok?

File: src/main/java/cinema/service/ShoppingCartServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    @Override
+    public void registerNewShoppingCart(User user) {
+        ShoppingCart shoppingCart = new ShoppingCart();
+        shoppingCart.setUser(user);
+        shoppingCartDao.add(shoppingCart);
+    }
+
+    @Override
+    public void clear(ShoppingCart shoppingCart) {
+        ShoppingCart clearingShoppingCart = shoppingCartDao.getByUser(shoppingCart.getUser());
```
Comment (KseniiaMakarova) [inline src/main/java/cinema/service/ShoppingCartServiceImpl.java]: do we need this?

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/cinema/service/ShoppingCartServiceImpl.java (hunk 4 above)
Comment (vicire) [reply src/main/java/cinema/service/ShoppingCartServiceImpl.java]: Fixed

File: src/main/java/cinema/service/ShoppingCartServiceImpl.java (hunk 3 above)
Comment (vicire) [reply src/main/java/cinema/service/ShoppingCartServiceImpl.java]: Fixed

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**